### PR TITLE
[fix](Nereids) fill miss slot in having subquery

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

fill miss slot in having subquery 
such as 
```
select * from t group by k having max(k) in (select k from t2)
```

the max(k) should be push down aggregate

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by morrySnow** [comment]
add case in desc

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubqueryToApply.java [hunk 1]
```diff
@@ -73,7 +74,8 @@ public List<Rule> buildRules() {
                     LogicalFilter<Plan> filter = ctx.root;
 
                     ImmutableList<Set<SubqueryExpr>> subqueryExprsList = filter.getConjuncts().stream()
-                            .<Set<SubqueryExpr>>map(e -> e.collect(SubqueryExpr.class::isInstance))
+                            .<Set<SubqueryExpr>>map(e ->
+                                    e.collect(subq -> subq instanceof SubqueryExpr && !(subq instanceof ListQuery)))
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubqueryToApply.java:78]: add some comments to explain why `e.collect(subq -> subq instanceof SubqueryExpr && !(subq instanceof ListQuery)))`

**Comment by XieJiann** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**